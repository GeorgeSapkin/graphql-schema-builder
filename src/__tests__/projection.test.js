'use strict';

const graphql = require('graphql');

const {
  getProjection
} = require('..')(graphql);

describe('getProjection', () => {
  describe('returns empty', () => {
    it('when fieldASTs is not set', () => {
      expect(getProjection()).toMatchSnapshot();
    });

    it('when fieldASTs.fieldNodes is not an array', () => {
      expect(getProjection({})).toMatchSnapshot();
    });

    it('when fieldASTs.fieldNodes is empty', () => {
      expect(getProjection({ fieldNodes: [] })).toMatchSnapshot();
    });

    it('when fieldASTs.fieldNodes[0].selectionSet is not set', () => {
      expect(getProjection({ fieldNodes: [{}] })).toMatchSnapshot();
    });

    it('when fieldASTs.fieldNodes[0].selectionSet.selections is not an array',
      () => {
        expect(getProjection({ fieldNodes: [{
          selectionSet: {}
        }] })).toMatchSnapshot();
      });
  });

  describe('returns', () => {
    it('an empty object', () => {
      expect(getProjection({ fieldNodes: [{
        selectionSet: {
          selections: []
        }
      }] })).toMatchSnapshot();
    });

    it('a projection object', () => {
      expect(getProjection({ fieldNodes: [{
        selectionSet: {
          selections: [{
            name: {
              value: 'a'
            }
          }, {
            name: {
              value: 'b'
            }
          }]
        }
      }] })).toMatchSnapshot();
    });
  });
});
