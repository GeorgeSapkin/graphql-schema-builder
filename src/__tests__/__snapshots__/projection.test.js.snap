// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProjection returns a projection object 1`] = `
Object {
  "a": 1,
  "b": 1,
}
`;

exports[`getProjection returns an empty object 1`] = `Object {}`;

exports[`getProjection returns empty when fieldASTs is not set 1`] = `Object {}`;

exports[`getProjection returns empty when fieldASTs.fieldNodes is empty 1`] = `Object {}`;

exports[`getProjection returns empty when fieldASTs.fieldNodes is not an array 1`] = `Object {}`;

exports[`getProjection returns empty when fieldASTs.fieldNodes[0].selectionSet is not set 1`] = `Object {}`;

exports[`getProjection returns empty when fieldASTs.fieldNodes[0].selectionSet.selections is not an array 1`] = `Object {}`;
